floss/language/go/extract.py

```diff
@@ -0,0 +1,388 @@
+    # Extract strings from string table in .rdata section
+    # .data:00537B40                 dd offset unk_4A1E3C
+    # .data:00537B44                 db    4
+    # .data:00537B45                 db    0
+    # .data:00537B46                 db    0
+    # .data:00537B47                 db    0
+    # .data:00537B48                 dd offset unk_4A21C2
+    # .data:00537B4C                 db    6
+    # .data:00537B4D                 db    0
+    # .data:00537B4E                 db    0
+    # .data:00537B4F                 db    0
+
+    for addr in range(0, len(section_data) - alignment // 2, alignment // 2):
+        try:
+            curr = section_data[addr : addr + alignment]
+            s_off, s_size = struct.unpack(fmt, curr)
+
+            if not s_off and not (1 <= s_size < 128):
+                continue
+
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+            if not pe.get_section_by_rva(s_rva):
+                continue
+
+            binary_string = pe.get_string_at_rva(s_rva, s_size)
+
+            addr = pe.get_offset_from_rva(s_rva)
+
+            yield from yield_string(binary_string, addr, min_length)
+
+        except Exception as e:
+            logger.error(f"Error: {e}")
+            raise
+
+
+def extract_longstrings(
+    pe: pefile.PE, section_data, section_va, min_length, pattern, arch, regex_offset: Optional[int] = 0
+) -> Iterable[StaticString]:
+    for m in pattern.finditer(section_data):
+        s_off = struct.unpack("<I", m.group("offset"))[0]
+        s_size = struct.unpack("<B", m.group("size"))[0]
+
+        if arch == "amd64":
+            s_rva = s_off + m.start() + section_va + regex_offset
+            try:
+                addr = pe.get_offset_from_rva(s_rva)
+            except pefile.PEFormatError:
+                continue
+
+        elif arch == "386":
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+            try:
+                addr = pe.get_offset_from_rva(s_rva)
+            except pefile.PEFormatError:
+                continue
+
+        binary_string = pe.get_string_at_rva(s_rva, s_size)
+        yield from yield_string(binary_string, addr, min_length)
+
+
+def extract_go_strings(
+    sample: pathlib.Path,
+    min_length,
+) -> Iterable[StaticString]:
+    """
+    Get Go strings from a PE file.
+    Reference: https://github.com/mandiant/flare-floss/issues/779
+    """
+
+    try:
+        pe = pefile.PE(sample)
+    except pefile.PEFormatError as err:
+        logger.debug(f"invalid PE file: {err}")
+        raise ValueError("Invalid PE header")
+
+    if pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_AMD64"]:
+        """
+        .data:0000000000770F20 3D 68 60 00 00 00+off_770F20      dq offset aString
+        .data:0000000000770F28 15                                db  15h
+        .data:0000000000770F29 00                                db    0
+        """
+        alignment = 0x10
+        arch = "amd64"
+        fmt = "<QQ"
```

Comment:
we can move all of this, except `arch`, now into `extract_string_blob_in_rdata_data`